pkg/test/resourcefixture/contexts/register.go

```diff
@@ -66,43 +71,57 @@ type ResourceContext struct {
 
 	// fields related to DCL-based resources
 	DCLSchema *openapi.Schema
-	DCLBased  bool
 }
 
 var (
 	resourceContextMap = map[string]ResourceContext{}
 	emptyGVK           = schema.GroupVersionKind{}
 )
 
-func GetResourceContext(fixture resourcefixture.ResourceFixture, serviceMetadataLoader dclmetadata.ServiceMetadataLoader, dclSchemaLoader dclschemaloader.DCLSchemaLoader) ResourceContext {
+func GetResourceContext(fixture resourcefixture.ResourceFixture, serviceMetadataLoader dclmetadata.ServiceMetadataLoader, dclSchemaLoader dclschemaloader.DCLSchemaLoader) (ResourceContext, error) {
 	rc, ok := resourceContextMap[fixture.Name]
 	if !ok {
 		rc = ResourceContext{
 			ResourceGVK:  fixture.GVK,
 			ResourceKind: fixture.GVK.Kind,
 		}
 	}
+
 	if rc.ResourceGVK == emptyGVK {
 		rc.ResourceGVK = fixture.GVK
 	}
+
+	// Determine the type of controller for the resource: TF, DCL, or Direct.
+	crd, err := crdloader.GetCRDForGVK(rc.ResourceGVK)
+	if err != nil {
+		return ResourceContext{}, err
+	}
+	if crd.GetLabels()[crdgeneration.TF2CRDLabel] == "true" {
```

Comment:
I think we've merged the SQLInstance clone support... should we be applying the predicates here?

(A fine answer is "yes, but first we want to fix a lot of the tests")

Reply:
yes, but first we want to fix a lot of the tests

Reply:
😄